AssertionError with Resolver (gmsk fail)

```Julia-repl
_
   _       _ _(_)_     |  Documentation: https://docs.julialang.org/
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.11.0-rc4 (2024-09-25)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.11) pkg> activate --temp
  Activating new project at `/var/folders/ds/_tfd5bfd0bn51v_5xqxy821h0000gn/T/jl_hyFIta`

(jl_hyFIta) pkg> st
Status `/private/var/folders/ds/_tfd5bfd0bn51v_5xqxy821h0000gn/T/jl_hyFIta/Project.toml` (empty project)

(jl_hyFIta) pkg> add ActuaryUtilities
   Resolving package versions...
┌ Warning: gmsk[32][326][19,63] fail
└ @ Pkg.Resolve ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Resolve/Resolve.jl:382
ERROR: AssertionError: verify_solution(sol, graph)
Stacktrace:
  [1] _resolve(graph::Pkg.Resolve.Graph, lower_bound::Nothing, previous_sol::Nothing)
    @ Pkg.Resolve ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Resolve/Resolve.jl:111
  [2] resolve
    @ ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Resolve/Resolve.jl:67 [inlined]
  [3] resolve_versions!(env::Pkg.Types.EnvCache, registries::Vector{Pkg.Registry.RegistryInstance}, pkgs::Vector{Pkg.Types.PackageSpec}, julia_version::VersionNumber, installed_only::Bool)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:424
  [4] targeted_resolve(env::Pkg.Types.EnvCache, registries::Vector{Pkg.Registry.RegistryInstance}, pkgs::Vector{Pkg.Types.PackageSpec}, preserve::Pkg.Types.PreserveLevel, julia_version::VersionNumber)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1384
  [5] tiered_resolve(env::Pkg.Types.EnvCache, registries::Vector{Pkg.Registry.RegistryInstance}, pkgs::Vector{Pkg.Types.PackageSpec}, julia_version::VersionNumber, try_all_installed::Bool)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1356
  [6] _resolve(io::IOContext{IO}, env::Pkg.Types.EnvCache, registries::Vector{Pkg.Registry.RegistryInstance}, pkgs::Vector{Pkg.Types.PackageSpec}, preserve::Pkg.Types.PreserveLevel, julia_version::VersionNumber)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1394
  [7] add(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}, new_git::Set{Base.UUID}; allow_autoprecomp::Bool, preserve::Pkg.Types.PreserveLevel, platform::Base.BinaryPlatforms.Platform, target::Symbol)
    @ Pkg.Operations ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1427
  [8] add
    @ ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1400 [inlined]
  [9] add(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; preserve::Pkg.Types.PreserveLevel, platform::Base.BinaryPlatforms.Platform, target::Symbol, allow_autoprecomp::Bool, kwargs::@Kwargs{io::IOContext{IO}})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/API.jl:306
 [10] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{IO}, kwargs::@Kwargs{})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/API.jl:159
 [11] add(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/API.jl:148
 [12] do_cmd(command::Pkg.REPLMode.Command, io::Base.TTY)
    @ Pkg.REPLMode ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/REPLMode/REPLMode.jl:407
 [13] do_cmds(commands::Vector{Pkg.REPLMode.Command}, io::Base.TTY)
    @ Pkg.REPLMode ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/src/REPLMode/REPLMode.jl:393
 [14] do_cmds(repl::REPL.LineEditREPL, commands::String)
    @ REPLExt ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:92
 [15] on_done(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool, repl::REPL.LineEditREPL)
    @ REPLExt ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:106
 [16] #invokelatest#2
    @ ./essentials.jl:1054 [inlined]
 [17] invokelatest
    @ ./essentials.jl:1051 [inlined]
 [18] (::REPLExt.var"#47#50"{REPL.LineEditREPL})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ REPLExt ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/Pkg/ext/REPLExt/REPLExt.jl:128
 [19] #invokelatest#2
    @ ./essentials.jl:1054 [inlined]
 [20] invokelatest
    @ ./essentials.jl:1051 [inlined]
 [21] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/REPL/src/LineEdit.jl:2749
 [22] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/REPL/src/REPL.jl:1456
 [23] (::REPL.var"#75#81"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL ~/.julia/juliaup/julia-1.11.0-rc4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.11/REPL/src/REPL.jl:461

julia> versioninfo()
Julia Version 1.11.0-rc4
Commit b4b9add84db (2024-09-25 11:03 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 8 × Apple M3
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, apple-m3)
Threads: 4 default, 0 interactive, 2 GC (on 4 virtual cores)
Environment:
  JULIA_NUM_THREADS = auto
```

It works if you specify a specific version (e.g. `add ActuaryUtiliites@5`). Julia installed via Juliaup and same issue also seen on a Windows machine on 1.10 and 1.11

--- a/src/Resolve/Resolve.jl
+++ b/src/Resolve/Resolve.jl
@@ -362,6 +362,8 @@ function greedysolver(graph::Graph)
                     return (false, Int[])
                 elseif old_v1 == spp[p1]
                     sol[p1] = v1
+                    fill!(gconstr[p1], false)
+                    gconstr[p1][v1] = true
                     push!(staged_next, p1)
                 end
             end

--- a/test/resolve.jl
+++ b/test/resolve.jl
@@ -617,7 +617,6 @@ end
     )
     @test resolve_tst(deps_data, reqs_data, want_data)
 
-
     # require A, D, and lower version of Y
     reqs_data = Any[
         ["A", "*"],
@@ -635,6 +634,39 @@ end
         "G"=>v"2",
         "H"=>v"1",
         "I"=>v"1",
+    )
+    @test resolve_tst(deps_data, reqs_data, want_data)
+
+
+    VERBOSE && @info("SCHEME 15")
+    ## DEPENDENCY SCHEME 15: A GRAPH WITH A WEAK DEPENDENCE
+    ## (REDUCED VERSION OF A REALISTIC SCHEME, ref Pkg.jl issue #4030)
+    deps_data = Any[
+        ["A", v"1"],
+        ["A", v"2", "C", "*"],
+        ["B", v"1", "D", "1", :weak],
+        ["C", v"1", "E", "*"],
+        ["C", v"2", "E", "*"],
+        ["C", v"2", "B", "1"],
+        ["E", v"1", "D", "1"],
+        ["E", v"2", "F", "1"],
+        ["F", v"1", "D", "*"],
+        ["D", v"1"],
+        ["D", v"2"],
+    ]
+
+    @test sanity_tst(deps_data)
+
+    reqs_data = Any[
+        ["A", "*"],
+    ]
+    want_data = Dict(
+        "A" => v"2",
+        "B" => v"1",
+        "C" => v"2",
+        "D" => v"1",
+        "E" => v"2",
+        "F" => v"1",
     )
     @test resolve_tst(deps_data, reqs_data, want_data)
 

